controllers/UserController.js

```diff
@@ -0,0 +1,59 @@
+import { UserService } from '../services';
+
+const signUp = async (req, res) => {
+  try {
+    const { nickname, email, password } = req.body;
+
+    //회원가입시 필요한 필수 키
+    const requiredKeys = { nickname, email, password };
+
+    const values = Object.values(requiredKeys);
+    const keys = Object.keys(requiredKeys);
+
+    const undefinedValuesOfKey = keys.filter((key) => {
+      return requiredKeys[key] === undefined;
+    });
+
+    if (values.includes(undefined)) {
+      let err = new Error(`KEY_ERROR: ${undefinedValuesOfKey}`);
+      err.statusCode = 400;
+      throw err;
+    } else {
+      await UserService.signUp(nickname, email, password);
+
+      res.status(201).json({ message: 'SIGNUP_SUCCESS', email });
```

Comment:
```suggestion
    const { nickname, email, password } = req.body;

    //회원가입시 필요한 필수 키
    const requiredKeys = ['nickname', 'email', 'password'];
    
    for (let key of requiredKeys) {
      if (!body[key]) {
        let err = new Error(`KEY_ERROR: ${key}`);
        err.statusCode = 400;
        throw err;
      }
    }
    await UserService.signUp(nickname, email, password);
    res.status(201).json({ message: 'SIGNUP_SUCCESS', email });
```
이정도로도 충분히 KEY_ERROR 발생시키실 수 있을 것 같은데, 어떻게 생각하시나요? 제 코드 혹시 이해되시나요?

참고로 `if` 문 내부에 return이 있어서 if 조건문 충족할 때, return으로 함수가 끝나버린다면,
불필요한 `else`문은 꼭 쓰지 않으셔도 됩니다 :) 보다 깔끔한 코드를 구현하실 수 있습니담

Reply:
헉 저런 방법이 있었군요! 어차피 제가 기존에 짰던 코드도 객체를 배열로 바꿔주는 역할도 하고 있으니까 복잡하게 하지 말고 저렇게 필수 키를 배열로 바로 넣어도 되었네요 끙... 
for of라는 루프문은 처음 보는데 너무 신기합니다🤔 리퀘스트 바디에서 필수key가 없을 때에 그에 해당하는 키에러를 쏴주는거라고 이해를 했는데 이게 맞을까욥...!🧐